hoot-core/src/main/cpp/hoot/core/io/OsmChangesetSqlFileWriter.cpp

```diff
@@ -170,13 +176,13 @@ ElementPtr OsmChangesetSqlFileWriter::_getChangeElement(ConstElementPtr element)
     default:
       throw HootException("Unknown element type");
   }
-
   return changeElement;
 }
 
 // If osm.changeset.file.writer.generate.new.ids is false, then these create methods assume
 // you've already set the ID correctly in terms of the OSM API target db for the element to be
 // created.
+//TODO: Can we get rid of the generate new ID's option now?
```

Comment:
Yeah, I guess in the xml writer workflow no calls to nextId should ever be needed as long as the changeset deriver correctly identifies all the create statements.  The nextId calls were being used to prevent id conflicts in the backend during conflation, but it would possibly be easier to just define the negative id generation for all secondary features instead.